multimodal/src/autogluon/multimodal/learners/ensemble.py

```diff
@@ -0,0 +1,719 @@
+
+            pred_file_path = os.path.join(per_learner_path, f"{mode}_predictions.npy")
+            if os.path.isfile(pred_file_path):
+                logger.info(f"{mode}_predictions.npy exists. loading it...")
+                y_pred = np.load(pred_file_path)
+            else:
+                if isinstance(per_learner, str):
+                    per_learner = BaseLearner.load(path=per_learner_path)
+                if not self._problem_type:
+                    self._problem_type = per_learner.problem_type
+                else:
+                    assert self._problem_type == per_learner.problem_type
+                outputs = per_learner.predict_per_run(
+                    data=data,
+                    realtime=False,
+                    requires_label=False,
+                )
+                y_pred = extract_from_output(outputs=outputs, ret_type=LOGITS)
+
+                if self._problem_type == REGRESSION:
+                    y_pred = per_learner._df_preprocessor.transform_prediction(y_pred=y_pred)
+                if self._problem_type in [BINARY, MULTICLASS]:
+                    y_pred = logits_to_prob(y_pred)
+                    if self._problem_type == BINARY:
+                        y_pred = y_pred[:, 1]
+
+                if save:
+                    np.save(pred_file_path, y_pred)
+
+            if requires_label:
+                label_file_path = os.path.join(per_learner_path, f"{mode}_labels.npy")
+                if os.path.isfile(label_file_path):
+                    logger.info(f"{mode}_labels.npy exists. loading it...")
+                    y_true = np.load(label_file_path)
+                else:
+                    if isinstance(per_learner, str):
+                        per_learner = BaseLearner.load(path=per_learner_path)
+                    y_true = per_learner._df_preprocessor.transform_label_for_metric(df=data)
+
+                    if save:
+                        np.save(label_file_path, y_true)
+
+                if labels is None:
+                    labels = y_true
+                else:
+                    assert np.array_equal(y_true, labels)
+
+            predictions.append(y_pred)
+
+        if requires_label:
+            return predictions, labels
+        else:
+            return predictions
+
+    @staticmethod
+    def verify_predictions_labels(predictions, learners, labels=None):
+        if labels is not None:
+            assert isinstance(labels, np.ndarray)
+        assert isinstance(predictions, list) and all(isinstance(ele, np.ndarray) for ele in predictions)
+        assert len(learners) == len(
+            predictions
+        ), f"len(learners) {len(learners)} doesn't match len(predictions) {len(predictions)}"
+
+    def fit_per_ensemble(
+        self,
+        predictions: List[np.ndarray],
+        labels: np.ndarray,
+    ):
+        weighted_ensemble = EnsembleSelection(
+            ensemble_size=self._ensemble_size * len(predictions),
```

Comment:
Why `* len(predictions)`? This probably isn't necessary and would slow down the training a lot. I'd recommend just always using a smallish value such as `ensemble_size=40`.

Reply:
len(predictions) is the number of models, not the number of samples. For example, if we train 10 models, then it would be 10. `ensemble_size` is a multiplier on this number, making the actual ensemble size adaptable to the number of models.

Reply:
Yeah I was aware of len(predictions) being the # of models, but I'll mention that in my testing with TabRepo, there is virtually zero benefit going beyond `ensemble_size=40`. In fact, going beyond this value is harmful due to validation overfitting on small datasets, and you are more likely to lose the sparseness property as you have very high ensemble_size, which will slow down your inference speed.

Refer to fig 4 here: https://github.com/autogluon/tabrepo/blob/main/data/sensitivity.png

Reply:
Similarly, it isn't necessary to use very small values. If I'm not mistaken, with 2 models the code currently only uses ensemble_size=4. This will be suboptimal, as 4 isn't enough to get good weights between the 2 models. I'd recommend always using between 25 and 40, but selecting a single value for simplicity. Tabular uses 40.

Reply:
Great suggestion! We used `2* model_num` in previous experiments. We can try 40 later to see whether it can improve performance.